actors/miner/tests/util.rs

```diff
@@ -204,8 +261,1024 @@ impl ActorHarness {
+                deadline.challenge,
+                entropy.to_vec(),
+                challenge_rand.clone(),
+            );
+
+            let mut proof_infos = Vec::with_capacity(infos.len());
+            for ci in infos {
+                let mut si = ci.clone();
+                if all_ignored.get(ci.sector_number) {
+                    si = good_info.clone().unwrap();
+                }
+                let proof_info = SectorInfo {
+                    proof: si.seal_proof,
+                    sector_number: si.sector_number,
+                    sealed_cid: si.sealed_cid,
+                };
+                proof_infos.push(proof_info);
+            }
+
+            let vi = WindowPoStVerifyInfo {
+                randomness: challenge_rand.clone(),
+                proofs: params.proofs.clone(),
+                challenged_sectors: proof_infos,
+                prover: RECEIVER_ID,
+            };
+            let exit_code = match cfg.verification_exit {
+                Some(exit_code) => exit_code,
+                None => ExitCode::Ok,
+            };
+            rt.expect_verify_post(vi, exit_code);
+        }
+
+        if cfg.expected_power_delta.is_some() {
+            let power_delta = cfg.expected_power_delta.unwrap();
+            let claim = UpdateClaimedPowerParams {
+                raw_byte_delta: power_delta.raw,
+                quality_adjusted_delta: power_delta.qa,
+            };
+            rt.expect_send(
+                *STORAGE_POWER_ACTOR_ADDR,
+                PowerMethod::UpdateClaimedPower as u64,
+                RawBytes::serialize(claim).unwrap(),
+                TokenAmount::from(0),
+                RawBytes::default(),
+                ExitCode::Ok,
+            );
+        }
+
+        rt.call::<Actor>(Method::SubmitWindowedPoSt as u64, &RawBytes::serialize(params).unwrap())
+    }
+
+    pub fn dispute_window_post(
+        &self,
+        rt: &mut MockRuntime,
+        deadline: &DeadlineInfo,
+        proof_index: u64,
+        infos: &Vec<SectorOnChainInfo>,
+        expect_success: Option<PoStDisputeResult>,
+    ) {
+        rt.set_caller(*ACCOUNT_ACTOR_CODE_ID, self.worker);
+        rt.expect_validate_caller_type(vec![*ACCOUNT_ACTOR_CODE_ID, *MULTISIG_ACTOR_CODE_ID]);
+
+        self.expect_query_network_info(rt);
+
+        let challenge_rand = Randomness(Vec::from([10, 11, 12, 13]));
+        let mut all_ignored = BitField::new();
+        let dln = self.get_deadline(rt, deadline.index);
+        let post = self.get_submitted_proof(rt, &dln, proof_index);
+
+        for idx in post.partitions.iter() {
+            let partition = self.get_partition_snapshot(rt, &dln, idx);
+            all_ignored |= &partition.faults;
+            assert!(partition.recoveries.is_empty());
+        }
+
+        // find the first non-faulty, non-skipped sector in poSt to replace all faulty sectors.
+        let mut maybe_good_info: Option<SectorOnChainInfo> = None;
```

Comment:
Can the code used to construct the post verify info be deduplicated?  I realize it's not in go actors but I think it could be.

Reply:
sure, thats a good idea.

Reply:
done, deduplicated with a `make_window_post_verify_info` method.